File: leetcodes/1. Two Sum/twoSum.py
def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """
    d = {}
    for i, n in enumerate(nums):
        m = target - n
        if m in d:
            return [d[m], i]
        else:
            d[n] = i


if __name__ == "__main__":
    import sys
    arg1 = [int(s) for s in sys.argv[1].split(',')]
    a = twoSum(arg1, int(sys.argv[2]))
    print(a)

File: leetcodes/1. Two Sum/twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        //input: number array, integer
        //output: number array
        /*
        1) bineray search or two pointers
           time O(nlogn), space O(1)

        2) use hashmap
           time O(n), space O(n)
        */
      if (nums == null || nums.Length < 2)
        return new int[2];
      Dictionary<int, int> dic = new Dictionary<int, int>();
      for (int i = 0; i < nums.Length; i ++) {
        if (dic.ContainsKey(target - nums[i]))
          return new int[]{i, dic[target-nums[i]]};
        else if (!dic.ContainsKey(nums[i]))
          dic.Add(nums[i], i);
      }
      return new int[2];

    }
}